frontend/packages/console-shared/src/hooks/usePinnedResources.ts

```diff
@@ -13,18 +13,55 @@ type PinnedResourcesType = {
 
 const PINNED_RESOURCES_CONFIG_MAP_KEY = 'console.pinnedResources';
 
+const getCustomizedPins = (id: string) => {
+  if (window.SERVER_FLAGS.perspectives) {
+    const perspectives: Perspective[] = JSON.parse(window.SERVER_FLAGS.perspectives);
+    const perspective = perspectives.find((p: Perspective) => p.id === id && p?.pinnedResources);
+    return perspective?.pinnedResources;
+  }
+  return null;
+};
+
 export const usePinnedResources = (): [string[], (pinnedResources: string[]) => void, boolean] => {
   const fireTelemetryEvent = useTelemetry();
   const [activePerspective] = useActivePerspective();
   const perspectiveExtensions = usePerspectives();
-  const defaultPins = useMemo(
+  const { findModel } = useModelFinder();
+  const pinsFromExtension: PinnedResourcesType = useMemo(
+    () =>
+      perspectiveExtensions.reduce(
+        (acc, e) => ({
+          ...acc,
+          [e.properties.id]: (e.properties.defaultPins || []).map((gvk) => {
+            const model = {
+              group: gvk.group,
+              version: gvk.version,
+              kind: gvk.kind,
+            };
+            return referenceForExtensionModel(model);
```

Comment:
Is `referenceForExtensionModel` confused when you pass gvk directly? For me it looks like gkv and model are the same here. Or maybe this was because of an TypeScript error because gkv as an optional kind now?

```suggestion
            return referenceForExtensionModel(gvk);
```